model/jpa/src/main/java/org/keycloak/models/jpa/entities/CompositeRoleEntity.java

```diff
@@ -0,0 +1,92 @@
+package org.keycloak.models.jpa.entities;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.QueryHint;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.hibernate.jpa.QueryHints;
+
+@Entity
+@IdClass(CompositeRoleEntityKey.class)
+@Table(name="COMPOSITE_ROLE", uniqueConstraints = {
+      @UniqueConstraint(columnNames = { "COMPOSITE", "CHILD_ROLE" })
+})
+@NamedQueries({
+    @NamedQuery(name="getChildrenRoleIds", hints = @QueryHint(name = QueryHints.HINT_READONLY, value = "true"),
+            query="select compositerole.childRoleId from CompositeRoleEntity compositerole where compositerole.compositeId = :roleId"),
+    @NamedQuery(name="getCompositeRolesByCompositeIds", hints = @QueryHint(name = QueryHints.HINT_READONLY, value = "true"),
+            query="select compositerole from CompositeRoleEntity compositerole where compositerole.compositeId in :roleIds"),
+    @NamedQuery(name="getChildRoleIdsForCompositeIds", hints = @QueryHint(name = QueryHints.HINT_READONLY, value = "true"),
+            query="select compositerole.childRoleId from CompositeRoleEntity compositerole where compositerole.compositeId in :roleIds"),
+    @NamedQuery(name="removeCompositeAndChildRoleEntry",
+            query="delete from CompositeRoleEntity compositerole where compositerole.compositeId = :compositeId and compositerole.childRoleId = :childId"),
+})
+public class CompositeRoleEntity implements Serializable {
+
+    private static final long serialVersionUID = 7375648337789837909L;
+    
+    public static String INSERT_NATIVE_QUERY = "insert into COMPOSITE_ROLE (COMPOSITE, CHILD_ROLE) values (?, ?)";
```

Comment:
This has to be adjusted for support of custom schema.

Reply:
Ah - good catch @hmlnarik . :-)
I suppose you refer to the need for adding the schema placeholder in the native query as described in https://docs.jboss.org/hibernate/orm/5.6/userguide/html_single/Hibernate_User_Guide.html#sql-global-catalog-schema, correct?

Reply:
@davoustp - in other places you might find `JpaUtils.getTableNameForNativeQuery("COMPOSITE_ROLE", em);` 

An updated snippet of code might look like: 

```
            String compositeRoleTable = JpaUtils.getTableNameForNativeQuery("COMPOSITE_ROLE", em);
            Query q = em.createNativeQuery("insert into " + compositeRoleTable + " (COMPOSITE, CHILD_ROLE) values (:composite, :child)")
                    .setParameter("composite", this.getId())
                    .setParameter("child", role.getId());
```

Reply:
@davoustp - I don't know ... I'm just referencing the way it is used in other places (as in the process of reviewing this PR if found this as well, but hadn't submitted it, yet)

Reply:
Ok, let's keep things consistent, then.
Updating.

Reply:
Done.
Test `ClientRegistrationTest` passing with Undertow, Wildfly and Quarkus.